core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -548,8 +548,11 @@ class KafkaZkClient private[zk] (zooKeeperClient: ZooKeeperClient,
    */
   def getAllTopicsInCluster(registerWatch: Boolean = false): Set[String] = {
     val getChildrenResponse = retryRequestUntilConnected(
```

Comment:
It looks like we should just call getChildren(TopicsZnode.path) since you add shouldPaginate check in getChildren method. Just to avoid duplicates.

Reply:
They're not quite duplicates; one returns a Seq and the other a Set, and one forces watch = true. Not blocking reuse, just a little more refactoring, but do we want to take the time? This code is known to work, anything else requires more manual testing.

Reply:
Got it. In that sense I'm ok with keeping it for now.

Reply:
I'll do a followup PR after this one is in.